agent/config/config_windows_test.go

```diff
@@ -134,3 +138,66 @@ func TestMemoryUnboundedWindowsDisabled(t *testing.T) {
 	assert.NoError(t, err)
 	assert.False(t, cfg.PlatformVariables.MemoryUnbounded.Enabled())
 }
+
+func TestGetConfigFileName(t *testing.T) {
+	configFileName := "/foo/bar/config.json"
+	testCases := []struct {
+		name             string
+		envVarVal        string
+		expectedFileName string
+		expectedError    error
+		cfgFileSid       string
+	}{
+		{
+			name:             "config file via env var, no errors",
+			envVarVal:        configFileName,
+			expectedFileName: configFileName,
+			expectedError:    nil,
+			cfgFileSid:       "",
+		},
+		{
+			name:             "default config file without env var, no errors",
+			envVarVal:        "",
+			expectedFileName: defaultConfigFileName,
+			expectedError:    nil,
+			cfgFileSid:       adminSid,
+		},
+		{
+			name:             "unable to validate cfg file error",
+			envVarVal:        "",
+			expectedFileName: "",
+			expectedError:    errors.New("Unable to validate cfg file"),
+			cfgFileSid:       "random-sid",
+		},
+		{
+			name:             "invalid cfg file error",
+			envVarVal:        "",
+			expectedFileName: "",
+			expectedError:    errors.New("Invalid cfg file"),
+			cfgFileSid:       "S-1-5-7",
+		},
+	}
+	defer func() {
+		osStat = os.Stat
```

Comment:
Can you use the `oswrapper` and its mock for the tests?

Reply:
i think this is simple enough no? i see other tests following the same logic.

Reply:
Personally I don't like the `oswrapper` because it introduces more complexity for something simple like this. Forst we need to introduce a new dependency, and second, the mentioned interface is not segregated and exposes too many methods that are not required. 
This is a valid approach, for example see the [sample code](https://github.com/adonovan/gopl.io/blob/master/ch11/echo/echo.go#L23) where they use this approach in [The Go Programming Language](https://www.gopl.io/) book.

Reply:
I've seen both approaches and I'm alright with either of them.